### PR TITLE
Add ability to remove a consumer

### LINKED ISSUE #26: Add ability to remove a consumer
From the Kafka explorer view, we should be able to select a consumer generated by vscode-kafka and remove it (it needs to be stopped before removing) 

See similar feature in conduktor:

Dec-15-2020 11-06-19

Removal fails if "the group is not empty"

### PR DESCRIPTION
**Pull request by angelozerr**
Add ability to remove a consumer

Fixes #26

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR provides the delete support from kafka explorer. 

Multiple delete is not supported (@fbricon  could we do that in an another PR?)

The PR is in draft since I need to clean my code.

File: package.json [hunk 1]
```diff
@@ -280,6 +281,12 @@
                     "light": "images/light/clear-all.svg",
                     "dark": "images/dark/clear-all.svg"
                 }
+            },
+            {
```
Comment (fbricon) [inline package.json:285]: unnecessary blank line

Comment (angelozerr) [reply package.json:285]: fixed

**Comment by angelozerr** [comment]
Just for information, I refactor pick methods to start every time with pickClient which picks the required Client (the cluster). We could in the future improve it to add in this method a select cluster step to select a cluster.

I need to write some documentation about delete feature.

**Comment by angelozerr** [comment]
> I need to write some documentation about delete feature.

Ok documentation has been written. You can review it.

**Comment by angelozerr** [comment]
Please note that if you try to delete a consumer group which is started, it will throw an error, I think we can have this behaviour for this PR but perhaps we should improve it like stopping the consumer before the delete.

We could stopping this consumer

 * without warning (stop + delete) according to a settings (ala forceStop)
 * with a warning (if you delete the consumer,the consumer will be stopped before)

@jlandersen have you some opinion about stopping consumer before the delete of consumer?

If we want to have this feature, we need to do some refactoring to manage link betwwena consumer group id and the started consumers linked to this consumer group.

**Comment by angelozerr** [comment]
Please note the error message is not very helpful when you try to delete a consumer which is started. I would like to improve that in the issue https://github.com/jlandersen/vscode-kafka/issues/108 (to cover other cases too)

File: docs/Explorer.md [hunk 2]
```diff
@@ -8,6 +8,24 @@ The Kafka explorer shows configured clusters with their topics, brokers, consume
 
 ![Screenshot-3](assets/screen-3.png)
 
+## Actions
+
+### Copy Label
+
+You can copy label of any tree node in the clipboard with `Keyboard` or `"Copy Label" contextual menu`:
```
Comment (fbricon) [inline docs/Explorer.md:15]: You can copy the label of any tree node in the clipboard with `Ctrl+C` (`Cmd+C` on Mac) or  via the `Copy Label` contextual menu:

File: docs/Explorer.md [hunk 3]
```diff
@@ -8,6 +8,24 @@ The Kafka explorer shows configured clusters with their topics, brokers, consume
...
+
+### Copy Label
+
+You can copy label of any tree node in the clipboard with `Keyboard` or `"Copy Label" contextual menu`:
+
+![Copy Label](assets/kafka-explorer-copylabel.png)
+
+### Delete
+
+You can delete cluster, topics and consumer groups with `Keyboard`, `Trashcan icon` or `"Delete" contextual menu`:
```
Comment (fbricon) [inline docs/Explorer.md:21]: You can delete clusters, topics and consumer groups with the `Delete` (`Cmd+Delete` on Mac) key, the `Trashcan` icon or `"Delete" contextual menu`:

File: docs/Explorer.md [hunk 4]
```diff
@@ -8,6 +8,24 @@ The Kafka explorer shows configured clusters with their topics, brokers, consume
...
+
+![Delete Consumer Group](assets/kafka-explorer-delete-consumergroup.png)
+
+Multiple delete is not supported for the moment. See [issue 107](https://github.com/jlandersen/vscode-kafka/issues/107).
+
+To delete a consumer group, this consumer must be stopped, otherwise the Delete action will report an error.
```
Comment (fbricon) [inline docs/Explorer.md:27]: To delete a consumer group, it first must be stopped, else the `Delete` action will report an error.

**Review by fbricon — CHANGES_REQUESTED** [review]
Works really well. But I think it really deserves #108, to be more user friendly

File: docs/Explorer.md (hunk 2 above)
Comment (angelozerr) [reply docs/Explorer.md]: fixed

File: docs/Explorer.md (hunk 3 above)
Comment (angelozerr) [reply docs/Explorer.md]: fixed

File: docs/Explorer.md (hunk 4 above)
Comment (angelozerr) [reply docs/Explorer.md]: fixed

**Comment by angelozerr** [comment]
> Works really well. But I think it really deserves #108, to be more user friendly

Indeed it's in my scope and other message error should be improved too.

I have created the issue https://github.com/jlandersen/vscode-kafka/issues/110 too to improve delete consumer group when there are consumers which are started.

**Comment by jlandersen** [comment]
Really happy to see this. Especially in dev/test scenarios where they creep up quite a bit.

@angelozerr I don't think it's super necessary to stop the consumers as part of this particular feature. It could just as well be a non-extension managed group. As long as we show some kind of acceptable error message I think that part is covered and the user can take action from there.

**Comment by angelozerr** [comment]
> Really happy to see this. Especially in dev/test scenarios where they creep up quite a bit.

Really happy you like it:)

> @angelozerr I don't think it's super necessary to stop the consumers as part of this particular feature. It could just as well be a non-extension managed group. As long as we show some kind of acceptable error message I think that part is covered and the user can take action from there.

Indeed I will improve error message in https://github.com/jlandersen/vscode-kafka/issues/108

The really annoying usecase is when you start a consumer and you close your vscode and you restart it. You will see again the consumer group, but you cannot stop it. I create a bug for that in https://github.com/jlandersen/vscode-kafka/issues/111

If we can stop a consumer which is not started by the opened vscode session, I agree that we must not stop the consumer when we wish to delete it.

**Pull request merged**